src/gt4py/next/otf/recipes.py

```diff
@@ -14,12 +14,39 @@
 from __future__ import annotations
 
 import dataclasses
-from typing import Any
 
 from gt4py.next.ffront import stages as ffront_stages
+from gt4py.next.iterator import ir as itir
 from gt4py.next.otf import stages, step_types, workflow
 
 
+@dataclasses.dataclass(frozen=True)
+class FieldopTransformWorkflow(workflow.NamedStepSequence):
+    """Modular workflow for transformations with access to intermediates."""
+
+    func_to_foast: workflow.SkippableStep[
+        ffront_stages.FieldOperatorDefinition | ffront_stages.FoastOperatorDefinition,
+        ffront_stages.FoastOperatorDefinition,
+    ]
+    foast_inject_args: workflow.Workflow[
+        ffront_stages.FoastOperatorDefinition, ffront_stages.FoastClosure
+    ]
+    foast_to_past_closure: workflow.Workflow[ffront_stages.FoastClosure, ffront_stages.PastClosure]
+    past_transform_args: workflow.Workflow[ffront_stages.PastClosure, ffront_stages.PastClosure]
+    past_to_itir: workflow.Workflow[ffront_stages.PastClosure, stages.ProgramCall]
+    foast_to_itir: workflow.Workflow[ffront_stages.FoastOperatorDefinition, itir.Expr]
+
+    @property
+    def step_order(self) -> list[str]:
```

Comment:
where is this used?

Reply:
In the superclass. It determines in which order the steps are run.

Reply:
In the superclass. It determines in which order the steps are run.